relayer-cli/src/utils/relay.ts

```diff
@@ -42,32 +40,45 @@ const relay = async (chainid: number, nonce: number) => {
   await txn.wait();
 };
 
-const relayBatch = async (chainid: number, nonce: number, iterations: number) => {
+const relayBatch = async (chainid: number, nonce: number, maxBatchSize: number, _contract: any) => {
   const routeParams = getBridgeConfig(chainid);
-
   const web3 = new Web3(routeParams.rpcOutbox);
-  const batchedSend = _batchedSend(web3, routeParams.rpcOutbox, process.env.PRIVATE_KEY, 0);
-
+  const batchedSend = _batchedSend(web3, routeParams.batcher, process.env.PRIVATE_KEY, 0);
   const contract = new web3.eth.Contract(_contract.abi, routeParams.veaOutbox);
-  const veaOutbox = getVeaOutboxArbToEth(routeParams.veaOutbox, process.env.PRIVATE_KEY, routeParams.rpcOutbox);
+  const veaOutbox = getVeaOutbox(routeParams.veaOutbox, process.env.PRIVATE_KEY, routeParams.rpcOutbox, chainid);
   const count = await getCount(veaOutbox, chainid);
 
-  let txns = [];
-
-  for (let i = 0; i < iterations; i++) {
-    const proof = await getProofAtCount(chainid, nonce + i, count);
-    const [to, data] = await getMessageDataToRelay(chainid, nonce + i);
-    txns.push({
-      args: [proof, nonce + i, to, data],
-      method: contract.methods.sendMessage,
-      to: contract.options.address,
-    });
+  while (nonce < count) {
+    let batchMessages = 0;
+    let txns = [];
+    while (batchMessages < maxBatchSize && nonce < count) {
+      const isMsgRelayed = await veaOutbox.isMsgRelayed(nonce);
+      if (isMsgRelayed) {
+        nonce++;
+        continue;
+      }
+      const proof = await getProofAtCount(chainid, nonce, count);
+      const [to, data] = await getMessageDataToRelay(chainid, nonce);
+      txns.push({
+        args: [proof, nonce, to, data],
+        method: contract.methods.sendMessage,
+        to: contract.options.address,
+      });
+      batchMessages += 1;
+      nonce++;
+    }
+    if (batchMessages > 0) {
+      try {
+        await batchedSend(txns);
+      } catch (error) {
```

Comment:
Ideally we do not try to use `try/catch` to save our applications from errors we are not expecting. If there is an unexpected error it means that the application is running in an unsafe state and should be shut down (and automatically restarted by the manager).

That also provides a clear unhealthy signal to the infrastructure, which can raise an alarm so we can check what is going on. Otherwise, the bot could be malfunctioning and nobody would notice until logs are checked or someone complains.

Reply:
The transactions in this context involve the `sendMessage` function in VeaOutbox, and failures might occur due to external contracts being called from `sendMessage`. I added a try/catch block to handle these specific cases because failing transactions would otherwise stop the bot entirely. 

Should I add separate logic for failing `sendMessage` trxns or that would be too much overhead?

Reply:
Usually, unexpected operations should in fact cause the application to exit, making it restart and causing an alarm. Otherwise, the bot would be operating in unknown conditions, which could make the scenario worse.

Bots should be prepared to recover from crashes or restarts, instead of trying to keep them alive no matter what.

We can discuss this in Slack so we have more context, but usually we should try to apply to bots the same things we did for the ping pong exercises.